### PR TITLE
Source Google Sheets: handle config errors

### LINKED ISSUE #25691: Categorize Google Sheets config errors
These errors should be recategorized as config errors:
- Missing required parameter "spreadsheetId"
- ('invalid_grant: Token has been expired or revoked.', {'error': 'invalid_grant', 'error_description': 'Token has been 
expired or revoked.'})

The error messages should also reflect the action we want users to take: 
- Missing required parameter "spreadsheetId"
The error message should read: The spreadsheet link is not valid. Enter the URL of the Google spreadsheet you want to sync. 

- ('invalid_grant: Token has been expired or revoked.', {'error': 'invalid_grant', 'error_description': 'Token has been 
expired or revoked.'})
The error message should read: Access to the spreadsheet expired or was revoked. Re-authenticate to restore access.

### PR DESCRIPTION
**Pull request by artem1205**
## What
Resolve https://github.com/airbytehq/airbyte/issues/25691

## How
change error message and type to `config_error`

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py [hunk 1]
```diff
@@ -56,18 +55,22 @@ def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
...
+            message = "Config error: "
             # Give a clearer message if it's a common error like 404.
             if err.resp.status == status_codes.NOT_FOUND:
-                reason = "Requested spreadsheet was not found."
-            logger.error(f"Formatted error: {reason}")
-            return AirbyteConnectionStatus(
-                status=Status.FAILED, message=f"Unable to connect with the provided credentials to spreadsheet. Error: {reason}"
-            )
+                message += "The spreadsheet link is not valid. Enter the URL of the Google spreadsheet you want to sync."
+            raise AirbyteTracedException(
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py:62]: Can we add unit tests.

**Review by lazebnyi — APPROVED** [review]
LGTM. Need add unit tests.

File: airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-google-sheets/source_google_sheets/source.py:62]: tests added

**Pull request merged**